src/riak_object.erl

```diff
@@ -129,6 +130,19 @@ new(B, K, V, MD) when is_binary(B), is_binary(K) ->
 newts(B, K, V, MD) ->
     new_int2(B, K, V, MD).
 
+-spec is_ts(riak_object()) -> {'true', pos_integer()} | 'false'.
+is_ts(RObj) ->
+    check_for_ddl(get_contents(RObj)).
+
+check_for_ddl([{Metadata, _V}]) ->
```

Comment:
Rethinking this now. The reason I originally wrote it that way was to avoid invoking `dict:find` on every object with siblings, so now we have a potentially significant performance hit on fullsync.